### PR TITLE
feat: watch/unwatch mute/unmute and report article

### PR DESCRIPTION
**Pull request by swkatmask**
https://mask.atlassian.net/browse/fw-1200
https://mask.atlassian.net/browse/fw-1201
https://mask.atlassian.net/browse/fw-1202

### CODE REVIEW

File: src/components/Actions/ArticleMuteButton.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        },
+    });
+    const loading = mutation.isPending;
+    return (
+        <MenuButton
+            {...rest}
+            onClick={async () => {
+                const identity = article.author.handle || article.author.id;
+                if (!muted) {
+                    const confirmed = await ConfirmModalRef.openAndWaitForClose({
```
Comment (guanbinrui) [inline src/components/Actions/ArticleMuteButton.tsx:35]:
```suggestion
                if (!muted) {
                    const identity = article.author.handle || article.author.id;
                    const confirmed = await ConfirmModalRef.openAndWaitForClose({
```

File: src/providers/firefly/SocialMedia.ts [hunk 2]
```diff
@@ -882,20 +921,79 @@ class FireflySocialMedia implements Provider {
...
+            });
+            if (!res.data) return false;
+            const address = id.toLowerCase();
+            const is_followed = res.data.find((x) => address === x.address.toLowerCase())?.is_followed;
+            return !!is_followed;
+        }
+        return false;
+    }
+
+    async reportSpamNFT(collectionId: string) {
```
Comment (guanbinrui) [inline src/providers/firefly/SocialMedia.ts:988]: It was removed in latest commit.

File: src/providers/firefly/SocialMedia.ts [hunk 3]
```diff
@@ -882,20 +921,79 @@ class FireflySocialMedia implements Provider {
...
+            type,
+        });
+        const response = await fireflySessionHolder.fetch<Response<void>>(url, { method: 'put' });
+        return response;
+    }
+    /**
+     * @param {WatchType} type
+     * @param {string} id - id for masx, userId for twtter, address for wallet
+     */
+    async unwatch(type: WatchType, id: string) {
```
Comment (guanbinrui) [inline src/providers/firefly/SocialMedia.ts:955]: Please define interface in `types/SocialMedia.ts`.

File: src/components/Actions/Article/ReportButton.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                            <Trans>Are you sure you want to report this article?</Trans>
+                        </div>
+                    ),
+                    variant: 'normal',
+                });
+                if (!confirmed) return;
+                try {
+                    await mutation.mutateAsync();
+                } catch (error) {
+                    enqueueErrorMessage(t`Failed to report @${article.title}`, { error });
```
Comment (guanbinrui) [inline src/components/Actions/Article/ReportButton.tsx:46]:
```suggestion
                    enqueueErrorMessage(t`Failed to report @${article.title}`, { error });
                    throw error;
```

File: src/providers/firefly/SocialMedia.ts [hunk 5]
```diff
@@ -728,31 +739,52 @@ class FireflySocialMedia implements Provider {
         // TODO Mocking result for now.
         return true;
     }
+    private async report_post(params: ReportPostParams) {
```
Comment (guanbinrui) [inline src/providers/firefly/SocialMedia.ts:742]:
```suggestion
    private async reportPost(params: ReportPostParams) {
```

File: src/providers/types/SocialMedia.ts [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import type { Pageable, PageIndicator } from '@masknet/shared-base';
 
 import type { BookmarkType, FireflyPlatform, RestrictionType, SocialSource } from '@/constants/enum.js';
+import type { WatchType } from '@/providers/types/Firefly.js';
```
Comment (guanbinrui) [inline src/providers/types/SocialMedia.ts:4]: Please define watch type for all social medias.

```ts
export enum WatchType {
    Profile = 'profile',
    Post = 'post',
    Address = 'address',
}
```

Comment (swkatmask) [reply src/providers/types/SocialMedia.ts:4]: https://firefly-api.readme.io/reference/followcontroller_follow The Firefly API only supports `maskx`, `wallet` and `twitter`

**Comment by swkatmask** [comment]
@guanbinrui  please take another look

File: src/providers/types/SocialMedia.ts [hunk 7]
```diff
@@ -28,6 +28,12 @@ export enum NotificationType {
     Act = 'act',
 }
 
+export enum WatchType {
+    Wallet = 'wallet',
+    MaskX = 'maskx',
+    Twitter = 'twitter',
+}
```
Comment (guanbinrui) [inline src/providers/types/SocialMedia.ts:35]:
```suggestion
export enum WatchType {
    Wallet = 'wallet',
    Profile = 'profile',
}
```

**Pull request merged**